File: LEETCODE/342.cpp
//342. Power of Four

class Solution {
public:
    bool isPowerOfFour(int n) {
        
        if(n==0)
            return false;
        if(n==1)
            return true;
        
        int count=0,x=n;
        if(n%4==0)
            while(n)
            {
                n/=4;
                count++;
            }
        
        int res=pow(4,count-1);
        if(res==x)
            return true;
        
        return false;
    }
};

File: LEETCODE/LEETCODE-TYPESCRIPT/342.ts
// 342. Power of Four
// https://leetcode.com/problems/power-of-four/

//Method-1 Brute force
function isPowerOfFour(n: number): boolean {
    if(n<=0){
        return false;
    }

    while(n%4===0){
        n/=4;
    }

    return n===1;
};

//Method-2 using log
function isPowerOfFour(n: number): boolean {
    if (n <= 0) {
        return false;
    }

    let x = Math.log(n) / Math.log(4);
    
    return Math.ceil(x) === Math.floor(x);
};